### DISCUSSION TITLE
PSR12 Line limit

### DISCUSSION THREAD

**Discussion by furai** [PHPCSStandards/PHP_CodeSniffer#348] [Q&A] [answered]
Hey,

I'm not sure why but the PSR12 rule for line limit where it should split them after 120 doesn't seem to be enforced?
https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/master/src/Standards/PSR12/ruleset.xml#L32-L40
I don't really understand this ruleset but for sure it doesn't work for me. According to the comment it should report lines longer than 120 as error? Or am I misunderstanding it?

Cheers,
Furai

**Comment by jrfnl** [comment, marked as answer]
@furai Are you by chance ignoring warnings ? As per PSR-12, this is a "soft" limit, i.e. a warning, not an error.

Ref: https://www.php-fig.org/psr/psr-12/#23-lines

**Reply by jrfnl** [reply]
Also see: https://github.com/PHPCSStandards/PHP_CodeSniffer/wiki/Customisable-Sniff-Properties#genericfileslinelength

**Reply by furai** [reply]
Oh, I didn't realise it was a warning and I had:
```
  <arg value="ns"/>
```
in my `phpcs.xml`. 

Can phpcbf fix line length warnings?

Thanks for help!

EDIT: OK, checked the docs and apparently it cannot.

**Reply by rodrigoprimo** [reply]
> Can phpcbf fix line length warnings?
> EDIT: OK, checked the docs and apparently it cannot.

@furai, just to confirm that indeed the line length warnings cannot be fixed by `phpcbf`.
